serializer/seata-serializer-kryo/src/main/java/io/seata/serializer/kryo/KryoSerializerFactory.java

```diff
@@ -48,6 +48,7 @@ public class KryoSerializerFactory {
         @Override
         protected Kryo create() {
             Kryo kryo = new Kryo();
+            kryo.setReferences(true);
```

Comment:
I m wondering that why kryo 4 the property of references is true, and kryo 5 is false. What is the reason for example like lower performance.

Reply:
Can you present a small test report?

Reply:
> Can you present a small test report?

This is not very clear
The kryo community didn't explain why

Reply:
> Can you present a small test report?

If not set, the stack will overflow
The scheme given by the community is set to true

Reply:
> If not set, the stack will overflow The scheme given by the community is set to true
yes, the stack will overflow. But this problem look like never happen in seata, because the object from database it can not with cycle, so why I m wonder the performance if set it true.

Reply:
> > If not set, the stack will overflow The scheme given by the community is set to true
> 
> yes, the stack will overflow. But this problem look like never happen in seata, because the object from database it can not with cycle, so why I m wonder the performance if set it true.

https://github.com/EsotericSoftware/kryo/issues/617
If complex objects are not involved, it is best not to enable them

Reply:
> > > If not set, the stack will overflow The scheme given by the community is set to true
> > 
> > 
> > yes, the stack will overflow. But this problem look like never happen in seata, because the object from database it can not with cycle, so why I m wonder the performance if set it true.
> 
> [EsotericSoftware/kryo#617](https://github.com/EsotericSoftware/kryo/issues/617) If complex objects are not involved, it is best not to enable them

4. X is enabled by default, and it needs to be tested whether it will have an impact after being closed

Reply:
> 4. X is enabled by default, and it needs to be tested whether it will have an impact after being closed

ok